src/DependencyInjection/SonataAdminExtension.php

```diff
@@ -249,4 +252,25 @@ private function replacePropertyAccessor(ContainerBuilder $container): void
         $modelChoice = $container->getDefinition('sonata.admin.form.type.model_choice');
         $modelChoice->replaceArgument(0, new Reference('form.property_accessor'));
     }
+
+    private function configureTwigTextExtension(ContainerBuilder $container, XmlFileLoader $loader, array $config): void
+    {
+        $bundles = $container->getParameter('kernel.bundles');
+
+        $container->setParameter('sonata.admin.configuration.legacy_twig_text_extension', $config['options']['legacy_twig_text_extension']);
+        $loader->load('twig_string.xml');
+
+        if (isset($bundles['SonataCoreBundle']) && false !== $config['options']['legacy_twig_text_extension']) {
+            $stringExtension = $container->getDefinition('sonata.string.twig.extension');
+            $stringExtension->replaceArgument(0, new Reference('sonata.core.twig.extension.text'));
+        } elseif (isset($bundles['SonataTwigBundle']) && false !== $config['options']['legacy_twig_text_extension']) {
+            if (
+                $container->hasDefinition('sonata.twig.extension.deprecated_text_extension')
```

Comment:
@wbloszyk I had to add this check in master. I think we should add it to SonataAdminBundle 3.x.
`deprecated_text_extension` does not exist in twig extension 1.x so the composer.json can't say we're supporting 1.x extension without this check.

WDYT ?

Reply:
https://github.com/sonata-project/SonataAdminBundle/commit/2fafa32f9d4c6cfb75a2b25c6e365d8bd5c9a8a4#diff-156ed93418fe946ee724987b790b24baR333-R341

It is in SonataAdminBundle 3.x

Reply:
I'm talking about the
```
$container->hasDefinition('sonata.twig.extension.deprecated_text_extension')
```

Without this line, it fails in master, which use form-extension 1.4.

Reply:
In master we should drop `legacy_twig_text_extension` and use `u` filter.

Reply:
I agree, but I was currently trying to make the quickest fix since the merge 3.x into master is already hard.

What I mean is 3.x branch is saying supporting form.extension ^1.3 version
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/composer.json#L32
Which is false.

We should either drop the 1.x support in 3.x branch. Either add my fix.

Reply:
Yea, I see it now. I made mistake. I check service existing by `get` not `has`. For now user must install CoreBundle (dependency) in Sonata 3. This bug is not in production. 

We should release fix with:
```php
if (isset($bundles['SonataCoreBundle']) && false !== $config['options']['legacy_twig_text_extension']) {
    $stringExtension = $container->getDefinition('sonata.string.twig.extension');
    $stringExtension->replaceArgument(0, new Reference('sonata.core.twig.extension.text'));
} elseif (isset($bundles['SonataTwigBundle']) && false !== $config['options']['legacy_twig_text_extension']) {
    if ($container->getDefinition('sonata.twig.extension.deprecated_text_extension')) {
        $stringExtension = $container->getDefinition('sonata.string.twig.extension');
        $stringExtension->replaceArgument(0, new Reference('sonata.twig.extension.deprecated_text_extension'));
    } else {
        if (class_exist('Twig\Extensions\TextExtension') {
            $stringExtension->replaceArgument(0, new Reference('sonata.twig.extension.deprecated_text_extension'));
             // some error_trigger - about deprecations
        } else {
            // expection about not installed twig/extensions  
        }
    }
}
```

Of course after refactoring.

Reply:
A fix with
```
if (isset($bundles['SonataCoreBundle']) && false !== $config['options']['legacy_twig_text_extension']) {
            $stringExtension = $container->getDefinition('sonata.string.twig.extension');
            $stringExtension->replaceArgument(0, new Reference('sonata.core.twig.extension.text'));
        } elseif (isset($bundles['SonataTwigBundle']) && false !== $config['options']['legacy_twig_text_extension']) {
            if ($container->hasDefinition('sonata.twig.extension.deprecated_text_extension')) {
                $stringExtension = $container->getDefinition('sonata.string.twig.extension');
                $stringExtension->replaceArgument(0, new Reference('sonata.twig.extension.deprecated_text_extension'));
            }
        }
```
isn't enough ?

Reply:
It can be little confusing when someone set `legacy_twig_text_extension` to true and it will be ignored.

Reply:
I see your point.

Adding an exception would be a BC-break.
A warning would be ok. How I do that ?

Reply:
Yes. 
`legacy_twig_text_extension` is set by default to true. Exception will be BC-break. Let's use `@trigger_error` too.

Reply:
> See my sugestion.

It's seems more clear to you than to me 😅
Do you want to open the PR with your suggestion and some `NEXT_MAJOR` comment ?

Reply:
I can do it tomorrow.